interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,374 @@
+    highBandHighLatency();
+    verySmallWindow();
+    assertEquals(0, testsFailed);
+  }
+
+  public void endTest() {
+    server.shutdown();
+  }
+
+  // It is necessary to restart the server every time in order to bring the window back down.
+  private void lowBandLowLatency() {
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(LOW_BAND, LOW_LAT);
+    endTest();
+  }
+
+  private void lowBandHighLatency() {
+    resetProxy(LOW_BAND, HIGH_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(LOW_BAND, HIGH_LAT);
+    endTest();
+  }
+
+  private void highBandLowLatency() {
+    resetProxy(HIGH_BAND, LOW_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(HIGH_BAND, LOW_LAT);
+    endTest();
+  }
+
+  private void highBandHighLatency() {
+    resetProxy(HIGH_BAND, HIGH_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(HIGH_BAND, HIGH_LAT);
+    endTest();
+  }
+
+  private void verySmallWindow() {
+    resetProxy(MED_BAND, MED_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, TINY_WINDOW);
+    doTest(MED_BAND, MED_LAT);
+    endTest();
+  }
+
+  /**
+   * Main testing method. Streams 4 MB of data from a server and records the final window and
+   * average bandwidth usage
+   *
+   * @param bandwidth
+   * @param latency
+   */
+  private void doTest(int bandwidth, int latency) {
+    // arbitrary. smallest stream size I tried that still left enough time to fully inflate the
+    // window
+    int streamSize = 4 * 1024 * 1024;
+
+    TestServiceGrpc.TestService stub = TestServiceGrpc.newStub(channel);
+    StreamingOutputCallRequest.Builder builder = StreamingOutputCallRequest.newBuilder();
+    builder.addResponseParameters(ResponseParameters.newBuilder().setSize(streamSize));
+    StreamingOutputCallRequest request = builder.build();
+
+    TestStreamObserver observer = new TestStreamObserver();
+    stub.streamingOutputCall(request, observer);
+    try {
+      observer.waitFor();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    // Don't actually need bwithUsed right now but keeping it around because it might be interesting
+    long bwithUsed = (streamSize / (observer.getElapsedTime() / TimeUnit.SECONDS.toNanos(1)));
+    long expectedWindow = (latency * (bandwidth / TimeUnit.SECONDS.toMillis(1))) * 2;
+    int lastWindow = logHandler.getLastWindow();
+
+    // deal with cases that either don't cause a window update or hit max window
+    expectedWindow = Math.min(MAX_WINDOW, (Math.max(expectedWindow, REGULAR_WINDOW)));
+
+    // I know this range looks large, but this allows for only one extra/missed window update
+    // (one extra update causes a 2x difference and one missed update causes a .5x difference)
+    if ((lastWindow < 2.1 * expectedWindow) && (expectedWindow < 2.1 * lastWindow)) {
+      System.out.println("PASS");
```

Comment:
This should throw an exception, not keep track of test counts.  JUnit already does this for you.